modules/apps/headless/headless-builder/headless-builder-web/src/main/resources/META-INF/resources/js/components/EditAPIApplication.tsx

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ r*
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayCard from '@clayui/card';
+import {Heading} from '@clayui/core';
+import ClayLayout from '@clayui/layout';
+import ClayModal from '@clayui/modal';
+import ClayNavigationBar from '@clayui/navigation-bar';
+import {fetch, openToast} from 'frontend-js-web';
+import React, {useEffect, useState} from 'react';
+
+import {APIApplicationManagementToolbar} from './APIApplicationManagementToolbar';
+import BaseAPIApplicationField from './baseComponents/BaseAPIApplicationFields';
+import {fetchJSON} from './utils/fetchUtil';
+import {getCurrentURLParamValue} from './utils/urlUtil';
+
+import '../../css/main.scss';
+
+interface EditAPIApplicationProps {
+	apiURLPaths: APIURLPaths;
+	portletId: string;
+}
+
+type ApplicationStatus = {
+	key: 'published' | 'unpublished';
+};
+
+type DataError = {
+	baseURL: boolean;
+	title: boolean;
+};
+
+type Data = {
+	actions: Actions;
+	applicationStatus: ApplicationStatus;
+	baseURL: string;
+	description: string;
+	title: string;
+	version: string;
+};
+
+const headers = new Headers({
```

Comment:
@cgmonte Those are the same headers defined in the fetchUtil file, right? You can just import them to avoid duplicated code.